channeldb/invoice_test.go

```diff
@@ -228,6 +280,44 @@ func TestInvoiceWorkflow(t *testing.T) {
 	}
 }
 
+// TestPayHashIndexFallback asserts that the invoice db properly prioritizes
+// lookups and updtes by payment address.
+func TestPayHashIndexFallback(t *testing.T) {
+	db, cleanUp, err := makeTestDB()
+	defer cleanUp()
+	assert.Nil(t, err)
+
+	// Add two random invoices.
+	invoice1, err := randInvoice(1000)
+	assert.Nil(t, err)
+
+	inv1Hash := invoice1.Terms.PaymentPreimage.Hash()
+	_, err = db.AddInvoice(invoice1, inv1Hash)
+	assert.Nil(t, err)
+
+	invoice2, err := randInvoice(2000)
+	assert.Nil(t, err)
+
+	inv2Hash := invoice2.Terms.PaymentPreimage.Hash()
+	_, err = db.AddInvoice(invoice2, inv2Hash)
+	assert.Nil(t, err)
+
+	// Now, query using invoice 1's payment address, but invoice 2's payment
+	// hash. We expect invoice 1 to be returned since the payment hash
+	// should only be considered as a fallback.
```

Comment:
This is also what the other comment about the sanity check refers to

Reply:
add consistency check 👍